gammapy/irf/psf_king.py

```diff
@@ -18,22 +22,30 @@ class PSFKing(object):
 
     Parameters
     ----------
+    energy_lo : `~astropy.units.Quantity`
+        Lower energy boundary of the energy bin.
+    energy_hi : `~astropy.units.Quantity`
+        Upper energy boundary of the energy bin.
     offset : `~astropy.coordinates.Angle`
         Offset nodes (1D)
-    energy : `~gammapy.utils.energy.Energy`
-        Energy nodes (1D)
     gamma : `~numpy.ndarray`
         PSF parameter (2D)
     sigma : `~astropy.coordinates.Angle`
         PSF parameter (2D)
     """
 
-    def __init__(self, offset, energy, gamma, sigma):
+    def __init__(self, energy_lo, energy_hi, offset, gamma, sigma, energy_thresh_lo=Quantity(0.1, 'TeV'),
+                 energy_thresh_hi=Quantity(100, 'TeV')):
+        self.energy_lo = energy_lo.to('TeV')
+        self.energy_hi = energy_hi.to('TeV')
         self.offset = Angle(offset)
-        self.energy = Energy(energy)
+        self.energy = np.sqrt(self.energy_lo * self.energy_hi)
```

Comment:
Just in case you didn't know, there is an ``EnergyBounds`` class in gammapy
https://github.com/gammapy/gammapy/blob/master/gammapy/utils/energy.py#L163

I would prefer all classes to accept and ``EnergyBounds`` instance as input in the constructor. If you want the LogBin center as in this case you can just call EnergyBounds.log_centers. But it's up to you.